contracts/InterestRate/StableRate.sol

```diff
@@ -0,0 +1,248 @@
+            _optimalStableLoanRate
+        );
+    }
+
+    /**
+     * @notice Update the parameters of the interest rate model (only callable by owner, i.e. Timelock)
+     * @param _baseRatePerBlockForStable The approximate target base APR, as a mantissa (scaled by BASE)
+     * @param _multiplierPerBlock The rate of increase in interest rate wrt utilization (scaled by BASE)
+     * @param _multiplierPerBlockForStable The rate of increase in interest rate wrt utilization (scaled by BASE)
+     * @param _jumpMultiplierPerBlockForStable The multiplierPerBlock after hitting a specified utilization point
+     * @param _kink The utilization point at which the jump multiplier is applied
+     * @param _stableRatePremium The multiplierPerBlock after hitting a specified utilization point
+     * @param _optimalStableLoanRate Optimal stable loan rate percentage.
+     */
+    function updateStableRateModel(
+        uint256 _baseRatePerBlockForStable,
+        uint256 _multiplierPerBlock,
+        uint256 _multiplierPerBlockForStable,
+        uint256 _jumpMultiplierPerBlockForStable,
+        uint256 _kink,
+        uint256 _stableRatePremium,
+        uint256 _optimalStableLoanRate
+    ) external virtual {
+        require(msg.sender == owner, "StableRateModel: only owner may call this function.");
+
+        updateStableRateModelInternal(
+            _baseRatePerBlockForStable,
+            _multiplierPerBlock,
+            _multiplierPerBlockForStable,
+            _jumpMultiplierPerBlockForStable,
+            _kink,
+            _stableRatePremium,
+            _optimalStableLoanRate
+        );
+    }
+
+    /**
+     * @notice Calculates the utilization rate of the market: `borrows / (cash + borrows - reserves)`
+     * @param cash The amount of cash in the market
+     * @param borrows The amount of borrows in the market
+     * @param reserves The amount of reserves in the market (currently unused)
+     * @return The utilization rate as a mantissa between [0, BASE]
+     */
+    function utilizationRate(
+        uint256 cash,
+        uint256 borrows,
+        uint256 reserves
+    ) public pure returns (uint256) {
+        // Utilization rate is 0 when there are no borrows
+        if (borrows == 0) {
+            return 0;
+        }
+
+        return (borrows * BASE) / (cash + borrows - reserves);
+    }
+
+    /**
+     * @notice Calculates the stable loan rate of the market: `stableborrows / totalborrows`
+     * @param stableborrows The amount of stable borrows in the market
+     * @param totalborrows The amount of total borrows in the market
+     * @return The stable loan rate as a mantissa between [0, BASE]
+     */
+    function stableLoanRate(uint256 stableborrows, uint256 totalborrows) public pure returns (uint256) {
+        // Loan rate is 0 when there are no stable borrows
+        if (totalborrows == 0) {
+            return 0;
+        }
+
+        return (stableborrows * BASE) / totalborrows;
```

Comment:
```suggestion
        return (stable borrows / totalBorrows) * BASE;
```
This is clearer because you are calculating the rate and then scaling it

Reply:
This can be done. If we divide stableBorrows with totalBorrows it will give decimal as result and solidity don not support it. so, to prevent this we have to multiple by Base before performing dividation.